src/test/java/io/cdap/plugin/successfactors/source/SuccessFactorsSourceTest.java

```diff
@@ -0,0 +1,467 @@
+    return new URL("http://localhost:8000/odata/v2/entity-name");
+  }
+
+  private SuccessFactorsResponseContainer getResponseContainer() {
+    return new SuccessFactorsResponseContainer(200, "ok",
+                                               "2.0", new byte[]{50});
+  }
+
+  private Schema getPluginSchema() throws IOException {
+    String schemaString = "{\"type\":\"record\",\"name\":\"SuccessFactorsColumnMetadata\",\"fields\":[{\"name\":" +
+      "\"backgroundElementId\",\"type\":\"long\"},{\"name\":\"bgOrderPos\",\"type\":\"long\"},{\"name\":" +
+      "\"description\",\"type\":[\"string\",\"null\"]},{\"name\":\"endDate\",\"type\":[{\"type\":\"long\"," +
+      "\"logicalType\":\"timestamp-micros\"},\"null\"]},{\"name\":\"lastModifiedDate\",\"type\":" +
+      "[{\"type\":\"long\",\"logicalType\":\"timestamp-micros\"},\"null\"]},{\"name\":\"project\",\"type\":" +
+      "\"string\"},{\"name\":\"startDate\",\"type\":[{\"type\":\"long\",\"logicalType\":\"timestamp-micros\"}," +
+      "\"null\"]},{\"name\":\"userId\",\"type\":\"string\"}]}";
+
+    return Schema.parseJson(schemaString);
+  }
+
+  @Test
+  public void testConfigurePipelineWSchemaNotNull() throws SuccessFactorsServiceException, TransportException,
+    IOException {
+    pluginConfigBuilder = SuccessFactorsPluginConfig.builder()
+      .referenceName("unit-test-ref-name")
+      .baseURL("http://localhost")
+      .entityName("entity-name")
+      .username("username")
+      .password("password");
+
+    pluginConfig = pluginConfigBuilder.build();
+    successFactorsTransporter = new SuccessFactorsTransporter(pluginConfig.getUsername(), pluginConfig.getPassword());
+    successFactorsUrlContainer = new SuccessFactorsUrlContainer(pluginConfig);
+    successFactorsSchemaGenerator = new SuccessFactorsSchemaGenerator(new SuccessFactorsEntityProvider(edm));
+
+    new Expectations(SuccessFactorsUrlContainer.class, SuccessFactorsTransporter.class,
+                     SuccessFactorsSchemaGenerator.class) {
+      {
+        successFactorsUrlContainer.getTesterURL();
+        result = getUrl();
+        minTimes = 0;
+
+        successFactorsUrlContainer.getMetadataURL();
+        result = getUrl();
+        minTimes = 0;
+
+        successFactorsTransporter.callSuccessFactorsEntity(null, anyString, anyString);
+        result = getResponseContainer();
+        minTimes = 0;
+
+        successFactorsSchemaGenerator.buildDefaultOutputSchema(anyString);
+        result = getPluginSchema();
+        minTimes = 0;
+      }
+    };
+
+    successFactorsSource = new SuccessFactorsSource(pluginConfig);
+    Map<String, Object> plugins = new HashMap<>();
+    MockPipelineConfigurer mockPipelineConfigurer = new MockPipelineConfigurer(null, plugins);
+    successFactorsSource.configurePipeline(pipelineConfigurer);
+    schema = pipelineConfigurer.getOutputSchema();
+    Assert.assertNotNull("Output Schema generated is not Null", schema);
```

Comment:
check the actual schema value, not just non-null

Reply:
Done.